internals-js/src/operations.ts

```diff
@@ -1261,7 +1287,11 @@ export class SelectionSet extends Freezable<SelectionSet> {
   clone(): SelectionSet {
     const cloned = new SelectionSet(this.parentType);
     for (const selection of this.selections()) {
-      cloned.add(selection.clone());
+      const clonedSelection = selection.clone();
```

Comment:
Notes to reviewer: another minor stuff that profiling showed up. I could probably have put those into a separate commit, but I was lazy. Sorry.